src/shims/foreign_items/posix/macos.rs

```diff
@@ -15,85 +16,102 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
-                let result = this.macos_readdir_r(args[0], args[1], args[2])?;
+                let &[dirp, entry, result] = check_args(args)?;
+                let result = this.macos_readdir_r(dirp, entry, result)?;
                 this.write_scalar(Scalar::from_i32(result), dest)?;
             }
             "ftruncate" => {
-                let result = this.ftruncate64(args[0], args[1])?;
+                let &[fd, length] = check_args(args)?;
+                let result = this.ftruncate64(fd, length)?;
                 this.write_scalar(Scalar::from_i32(result), dest)?;
             }
 
             // Environment related shims
             "_NSGetEnviron" => {
+                let &[] = check_args(args)?;
                 this.write_scalar(this.machine.env_vars.environ.unwrap().ptr, dest)?;
             }
 
             // Time related shims
             "gettimeofday" => {
-                let result = this.gettimeofday(args[0], args[1])?;
+                let &[tv, tz] = check_args(args)?;
+                let result = this.gettimeofday(tv, tz)?;
                 this.write_scalar(Scalar::from_i32(result), dest)?;
             }
             "mach_absolute_time" => {
+                let &[] = check_args(args)?;
                 let result = this.mach_absolute_time()?;
                 this.write_scalar(Scalar::from_u64(result), dest)?;
             }
 
             "mach_timebase_info" => {
-                let result = this.mach_timebase_info(args[0])?;
+                let &[info] = check_args(args)?;
+                let result = this.mach_timebase_info(info)?;
                 this.write_scalar(Scalar::from_i32(result), dest)?;
             },
 
             // Access to command-line arguments
             "_NSGetArgc" => {
+                let &[] = check_args(args)?;
                 this.write_scalar(this.machine.argc.expect("machine must be initialized"), dest)?;
             }
             "_NSGetArgv" => {
+                let &[] = check_args(args)?;
                 this.write_scalar(this.machine.argv.expect("machine must be initialized"), dest)?;
             }
 
             // Thread-local storage
             "_tlv_atexit" => {
-                let dtor = this.read_scalar(args[0])?.not_undef()?;
+                let &[dtor, data] = check_args(args)?;
+                let dtor = this.read_scalar(dtor)?.not_undef()?;
                 let dtor = this.memory.get_fn(dtor)?.as_instance()?;
-                let data = this.read_scalar(args[1])?.not_undef()?;
+                let data = this.read_scalar(data)?.not_undef()?;
                 let active_thread = this.get_active_thread()?;
                 this.machine.tls.set_macos_thread_dtor(active_thread, dtor, data)?;
             }
 
             // Querying system information
             "pthread_get_stackaddr_np" => {
-                let _thread = this.read_scalar(args[0])?.not_undef()?;
+                let &[_thread] = check_args(args)?;
```

Comment:
The `_thread` is not unused... I think it's better not to start it with an `_` then. That way we get a warning if we fail to at least do a basic `read_scalar` on it.